cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/UninstallUtils.java

```diff
@@ -0,0 +1,121 @@
+        }
+        orgName = fullPkgPath.substring(0, orgNameIndex);
+
+        // Get package name
+        int packageNameIndex = fullPkgPath.indexOf(":");
+        if (packageNameIndex == -1) { // version is not provided
+            throw LauncherUtils.createLauncherException("error: no package version is provided");
+        }
+        packageName = fullPkgPath.substring(orgNameIndex + 1, packageNameIndex);
+        version = fullPkgPath.substring(packageNameIndex + 1, fullPkgPath.length());
+
+        Path homeRepoPath = BALLERINA_HOME_PATH.resolve(ProjectDirConstants.DOT_BALLERINA_REPO_DIR_NAME);
+        Path cacheRepoPath = BALLERINA_HOME_PATH.resolve(ProjectDirConstants.CACHES_DIR_NAME)
+                                                .resolve(ProjectDirConstants.BALLERINA_CENTRAL_DIR_NAME);
+        Path pkgDirPath = Paths.get(orgName, packageName, version);
+
+        // Check if package is installed locally
+        if (Files.exists(homeRepoPath.resolve(pkgDirPath), LinkOption.NOFOLLOW_LINKS)) {
+            deletePackage(homeRepoPath, pkgDirPath, fullPkgPath, packageName);
+        } else if (Files.exists(cacheRepoPath.resolve(pkgDirPath), LinkOption.NOFOLLOW_LINKS)) {
+            deletePackage(cacheRepoPath, pkgDirPath, fullPkgPath, packageName);
+        } else {
+            // package to be uninstalled doesn't exists
+            throw LauncherUtils.createLauncherException("error: incorrect package signature provided " + fullPkgPath);
+        }
+    }
+
+    /**
+     * Remove package from home repository.
+     *
+     * @param repoPath    path to the repository which contains the installed and pulled packages
+     * @param pkgDirPath  package directory path
+     * @param fullPkgPath full package path as given by user which is used for logging purposes
+     * @param pkgName     package name
+     */
+    private static void deletePackage(Path repoPath, Path pkgDirPath, String fullPkgPath, String pkgName) {
+        try {
+            Path path = repoPath.resolve(pkgDirPath);
+            // Delete the package zip
+            Files.deleteIfExists(path.resolve(pkgName + ProjectDirConstants.BLANG_COMPILED_PKG_EXT));
+
+            // Delete the empty directories
+            deleteEmptyParentDirs(path, repoPath);
+
+            // Print that the package was successfully uninstalled
+            outStream.println(fullPkgPath + " successfully uninstalled");
+        } catch (IOException e) {
+            throw LauncherUtils.createLauncherException("error: uninstalling package " + fullPkgPath + " was " +
+                                                                "unsuccessful");
+        }
+    }
+
+    /**
+     * Delete empty parent directories.
+     *
+     * @param from package directory path
+     * @param to   home repository path
+     * @throws IOException throw an exception if an error occurs
+     */
+    private static void deleteEmptyParentDirs(Path from, Path to) throws IOException {
```

Comment:
I don't feel this method is designed correctly or the params names are misleading. Lets have a chat on this